src/cocotb/share/lib/vpi/VpiImpl.cpp

```diff
@@ -157,20 +157,26 @@ static gpi_objtype_t to_gpi_objtype(int32_t vpitype) {
 }
 
 static gpi_objtype_t const_type_to_gpi_objtype(int32_t const_type) {
+    // Most simulators only return vpiDecConst or vpiBinaryConst
     switch (const_type) {
 #ifdef IUS
         case vpiUndefined:
             LOG_WARN(
                 "VPI: Xcelium reports undefined parameters as vpiUndefined, "
                 "guessing this is a logic vector");
-            return GPI_NET;
+            return GPI_REGISTER;
 #endif
```

Comment:
I see questa is also failing the test_discovery test that led me to this, should we maybe change the vpi_get(vpiVector) in the GPI_REGISTER code?